pkg/middleware/auth/middleware.go

```diff
@@ -202,3 +230,58 @@ func getReadableCertTypeName(certTypeID string) string {
 	}
 	return certTypeID
 }
+
+// buildResponsePayload creates a BRC-103 response payload according to BRC-104 section 6.9
+func buildResponsePayload(requestID string, statusCode int, headers http.Header, body []byte) ([]byte, error) {
+	writer := util.NewWriter()
+	requestIDBytes, err := base64.StdEncoding.DecodeString(requestID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid request ID format: %w", err)
+	}
+	if len(requestIDBytes) != 32 {
+		return nil, fmt.Errorf("request ID must be 32 bytes, got %d", len(requestIDBytes))
+	}
+	writer.WriteBytes(requestIDBytes)
+	statusCodeUInt, err := to.UInt64(statusCode)
+	if err != nil {
+		return nil, fmt.Errorf("invalid status code: %w", err)
+	}
+	writer.WriteVarInt(statusCodeUInt)
+	includedHeaders := filterAndSortResponseHeaders(headers)
+	writer.WriteVarInt(uint64(len(includedHeaders)))
+	for _, header := range includedHeaders {
+		keyBytes := []byte(header[0])
+		writer.WriteVarInt(uint64(len(keyBytes)))
+		writer.WriteBytes(keyBytes)
+		valueBytes := []byte(header[1])
+		writer.WriteVarInt(uint64(len(valueBytes)))
+		writer.WriteBytes(valueBytes)
+	}
+	if len(body) > 0 {
+		writer.WriteVarInt(uint64(len(body)))
+		writer.WriteBytes(body)
+	} else {
+		writer.WriteVarInt(uint64(0))
+	}
```

Comment:
```suggestion
	writer.WriteVarInt(uint64(len(body)))
        if len(body) > 0 {
	    writer.WriteBytes(body)
        }
```
just always write the body length first, then conditionally write bytes only if present... think its more cleaner :)

Reply:
changed